### PR TITLE
Show switch

### PR DESCRIPTION
**Pull request by shwsun**
This PR addresses part of the #572 's need. Per discussion with @henn and @gsilvis , right now we have `show_switch` api call, however we still need support from the switch backend. 

- [x] The commits are consistent with our contribution guidelines
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (tests/unit, pep8)
- [x] You've updated the documentation (if adding/changing api calls)

One quick question is: do we want a auth test for it? It doesn't seem like we have one for `list_switches`.

### CODE REVIEW

File: haas/api.py [hunk 1]
```diff
@@ -731,6 +731,28 @@ def switch_delete_port(switch, port):
...
+    """Show details of a switch.
+
+    Returns a JSON object regrading the switch. See `docs/rest_api.md`
+    for a full description of the output.
+
+    FIXME: Ideally this api call should return all detail information about
+    this switch. right now it needs support from the switch backend.
+    """
+    switch = _must_find(model.Switch, switch)
+    get_auth_backend().require_admin()
```
Comment (gsilvis) [inline haas/api.py]: The authorization check should be run /before/ anything that might return or throw an exception.

File: tests/unit/api/main.py [hunk 2]
```diff
@@ -1343,6 +1343,23 @@ def test_list_switches(self):
...
+        assert json.loads(api.show_switch('sw0')) == {
+            'name': 'sw0',
+            'ports': [{'label': '3'}]
+        }
+
+        api.switch_register_port('sw0', 'test_port')
+        assert json.loads(api.show_switch('sw0')) == {
+            'name': 'sw0',
+            'ports': [{'label': '3'},
+                      {'label': 'test_port'}]
```
Comment (shwsun) [inline tests/unit/api/main.py]: I agree with that, however I am not that concerned. The test cases are fixed and we will obey the orders, besides I don't think there is a good solution to sort the ports without breaking the json object.

File: haas/api.py (hunk 1 above)
Comment (shwsun) [inline haas/api.py]: Oops, will fix it.

File: tests/unit/api/main.py (hunk 2 above)
Comment (gsilvis) [inline tests/unit/api/main.py]: Just making a comment about this issue is fine too

File: haas/cli.py [hunk 3]
```diff
@@ -679,6 +679,13 @@ def list_project_networks(project):
 
 
 @cmd
+def show_switch(switch):
+    """Display information about <network>"""
```
Comment (knikolla) [inline haas/cli.py]: Docstring references network instead of switch.

File: tests/unit/api/main.py [hunk 4]
```diff
@@ -1343,6 +1343,25 @@ def test_list_switches(self):
         ]
 
 
+class Test_show_switch:
```
Comment (gsilvis) [inline tests/unit/api/main.py]: CamelCase is the standard for classes

**Comment by knikolla** [comment]
@shwsun can you update the branch?

**Comment by gsilvis** [comment]
Squashing the commits too might be nice

**Pull request merged**